FSA: Add DEVELOPMENT.md (#5076)
awsdocs/aws-doc-sdk-examples: applications/feedback_sentiment_analyzer/DEVELOPMENT.md

# FSA development

> This document describes the process for adding a new set of AWS Lambda functions for use with the Feedback Sentiment Analyzer (FSA). The requirements for each individual Lambda function are in [the specification](./SPECIFICATION.md).

Reviewer: I would not say they are a "new set of AWS Lambda functions". Rather, they are the same functions implemented in a new language.

Author: I take your first point, but they don't have to be in a new language. A user could make it in an existing language if they wanted to. I'll noodle on the wording.

For point 2, yes, I'll address that.

> This document describes the process for adding an additional set of AWS Lambda functions for use with the Feedback Sentiment Analyzer (FSA). The requirements for each individual Lambda function are in [SPECIFICATION.md](./SPECIFICATION.md).

Reviewer: Not new functions as far as the reader is concerned. Same functions with new language implementation.

Reviewer: ```suggestion
This document explains how to implement the AWS Lambda functions detailed in [SPECIFICATION.md](./SPECIFICATION.md#lambda-functions).
```

Author: Implement isn't the correct word here, maybe integrate?
```suggestion
This document explains how to integrate the AWS Lambda functions detailed in [SPECIFICATION.md](./SPECIFICATION.md#lambda-functions) in your SDK of choice.
```

};
```

2. `JAVASCRIPT_FUNCTIONS` is undefined. Define a constant named `JAVASCRIPT_FUNCTIONS` and set it equal to an empty `AppFunctionConfig` array.

```typescript
const JAVASCRIPT_FUNCTIONS: AppFunctionConfig[] = [];
```

3. Define the objects inside the `JAVASCRIPT_FUNCTIONS` array. Each object is an [AppFunctionConfig](./cdk/lib/constructs/app-lambdas.ts#4) which extends [FunctionProps](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FunctionProps.html) from the AWS Cloud Development Kit (AWS CDK) library. FSA requires four named Lambda functions:

- ExtractText
- AnalyzeSentiment
- TranslateText
- SynthesizeAudio

A definition for the `ExtractText` function might look like the following:

```typescript
  {
    ...BASE_APP_FUNCTION,
    codeAsset() {
      return Code.fromInline(`
        exports.handler = async (event) => {
          // implementation goes here
        }
    `);
    },
    name: "ExtractText",
  },
```

The `codeAsset` function must return a [Code](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Code.html) object. You can include inline code or [create a bundled asset](#bundling).

## Implement Lambda functions

> The following sections provide a general overview of the required Lambda functions. The accompanying CDK script orchestrates these functions as [AWS Step Functions](https://docs.aws.amazon.com/step-functions/latest/dg/welcome.html). For a detailed specification of these functions see [SPECIFICATION.md](./SPECIFICATION.md).

Reviewer: Here you say SPECIFICATION.md but above you say "the specification". Let's pick one and keep it consistent.

Author: Fixed

> ### ExtractText

Reviewer: Link back to ExtractText function in SPECIFICATION.md.

Extract the text from an image using Amazon Textract.

> ### AnalyzeSentiment

Reviewer: Link back to AnalyzeSentiment function in SPECIFICATION.md.

Pass the extracted text from ExtractText to Amazon Comprehend to determine the sentiment and the source language.

> ### TranslateText

Reviewer: Link back to TranslateText function in SPECIFICATION.md.

Translate the text to the language of your choice using Amazon Translate. The provided functions translate to French by default.

> ### SynthesizeAudio

Reviewer: Link back to SynthesizeAudio function in SPECIFICATION.md.

> Synthesize an audio file from the translated text using Amazon Polly.

Reviewer: ```suggestion
Synthesize an audio file from the translated text using Amazon Polly. See [function specification](SPECIFICATION.md#link-to-function-spec).
```

> ## Bundling

Reviewer: Is bundling required? If not:
```suggestion
## Bundling (optional)
```
